### PR TITLE
add system transaction generation test cases

### PR DESCRIPTION
**Pull request by EricBlack**
Test cases about GenerateSystemTransactions and cover SignAsync logic for system transaction

### CODE REVIEW

File: AElf.Kernel.Core.Tests/AElf.Kernel.Core.Tests.csproj [hunk 1]
```diff
@@ -18,6 +18,7 @@
     </ItemGroup>
 
     <ItemGroup>
+        <ProjectReference Include="..\AElf.Kernel.Consensus\AElf.Kernel.Consensus.csproj" />
```
Comment (loning) [inline AElf.Kernel.Core.Tests/AElf.Kernel.Core.Tests.csproj]: don't reference this project

Comment (EricBlack) [reply AElf.Kernel.Core.Tests/AElf.Kernel.Core.Tests.csproj]: removed this reference, thx~

File: AElf.Kernel.Core.Tests/KernelCoreTestAElfModule.cs [hunk 2]
```diff
@@ -19,15 +16,33 @@ namespace AElf.Kernel
...
+            
+            services.AddTransient(provider =>
+            {
+                var transactionList = new List<Transaction>
+                {
+                    new Transaction() {From = Address.Zero, To = Address.Generate(), MethodName = "InValue"},
+                    new Transaction() {From = Address.Zero, To = Address.Generate(), MethodName = "OutValue"},
+                };
+                var consensusTransactionGenerator = new Mock<ISystemTransactionGenerator>();
+                consensusTransactionGenerator.Setup(m => m.GenerateTransactions(@It.IsAny<Address>(), It.IsAny<long>(),
```
Comment (loning) [inline AElf.Kernel.Core.Tests/KernelCoreTestAElfModule.cs]: `@It`

File: AElf.Kernel.Core.Tests/KernelCoreTestAElfModule.cs [hunk 3]
```diff
@@ -19,15 +16,33 @@ namespace AElf.Kernel
...
+                        It.IsAny<Hash>(), ref It.Ref<List<Transaction>>.IsAny))
+                    .Callback(
+                        new MockGenerateTransactions((Address @from, long preBlockHeight, Hash previousBlockHash,
```
Comment (loning) [inline AElf.Kernel.Core.Tests/KernelCoreTestAElfModule.cs]: `@from` => fromAddress

File: AElf.Kernel.Core.Tests/KernelCoreTestAElfModule.cs (hunk 2 above)
Comment (EricBlack) [reply AElf.Kernel.Core.Tests/KernelCoreTestAElfModule.cs]: Removed all '@' symbols in test, sorry for did't understand the symbol meaning used with parameter.

**Pull request merged**